cogfiles/reaction_roles.py

```diff
@@ -0,0 +1,139 @@
+        self.bot = bot
+        self.reactionroles: dict[int, set[ReactionRole]] = {}
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        """Initialize the list of reaction roles in memory."""
+        await initialize_from_json(self.bot, self.reactionroles, ReactionRole, "reaction_roles")
+    
+    @app_commands.command()
+    @app_commands.rename(emoji_str="emoji")
+    @app_commands.checks.has_permissions(manage_guild=True)
+    async def reactionrole(self, interaction: discord.Interaction, role: discord.Role, emoji_str: str, message_link: str):
+        "Attaches a reaction role to a message. When a user reacts to the message with the given emoji, they will receive the role."
+        emoji = discord.PartialEmoji.from_str(emoji_str)
+
+        regex_searches = re.search("(\d+)\/(\d+)$", message_link)
+        try:
+            channel_id = regex_searches.group(1)
+            message_id = regex_searches.group(2)
+            channel = await interaction.guild.fetch_channel(int(channel_id))
+            message = await channel.fetch_message(int(message_id))
+        except:
+            return await interaction.response.send_message("Could not find the given message.", ephemeral=True)
+
+        try:
+            await message.add_reaction(emoji)
+        except discord.errors.HTTPException as e:
+            return await interaction.response.send_message("Could not use the given emoji. Make sure that this bot shares a server with the emoji.", ephemeral=True)
+
+        reactionrole = ReactionRole(channel, message, role, emoji)
+
+        # If there already exists a reaction role for this role, remove it and replace it with the new one
+        old_reactionrole = next((rr for rr in self.reactionroles[interaction.guild.id] if rr.role == reactionrole.role), None)
+        self.reactionroles[interaction.guild.id].discard(old_reactionrole)
+        self.reactionroles[interaction.guild.id].add(reactionrole)
+        write_json(interaction.guild.id, "reaction_roles", value=[reactionrole.to_json() for reactionrole in self.reactionroles[interaction.guild.id]])
+
+        embed = RandomColorEmbed(title="Reaction Role", description=f"React to this message with {emoji} to get the {role.mention} role: {message.jump_url}")
+        if old_reactionrole is not None:
+            embed.description += f"\nThis command replaced an existing reaction role for {role.mention} at {old_reactionrole.message.jump_url}"
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+
+    @app_commands.command()
+    @app_commands.checks.has_permissions(manage_guild=True)
+    async def reactionroleremove(self, interaction: discord.Interaction, role: discord.Role):
+        "Removes a reaction role message."
+        # If there already exists a reaction role for this role, remove it and replace it with the new one
+        reactionrole = next((rr for rr in self.reactionroles[interaction.guild.id] if rr.role == role), None)
```

Comment:
Similar to above, maybe rename `rr` to `role` and `reactionrole` to `existing_reactionrole`.